checkov/gitlab_ci/checks/job/AvoidDoublePipelines.py

```diff
@@ -20,10 +20,11 @@ def __init__(self) -> None:
 
     def scan_conf(self, conf: dict[str, Any]) -> tuple[CheckResult, dict[str, Any]]:
         c = 0
-        for rule in conf:
-            if isinstance(rule, dict) and "if" in rule:
-                value = rule['if']
-                if value.startswith('$CI_PIPELINE_SOURCE'):
+        pipeline_sources = ('$CI_PIPELINE_SOURCE == "merge_request_event"', '$CI_PIPELINE_SOURCE == "push"')
+        for x in conf:
+            if "if" in x:
+                value = x['if']
```

Comment:
@eurogig Since we don't know the type of `x`, let's add an ignore comment to mypy here:
```suggestion
                value = x['if']. # type:ignore[index]
```

Reply:
fixed it 🙂 just needed an `isinstance(x, dict)`, I also renamed `x` to `rule` 😄